str0m/src/media/mline.rs

```diff
@@ -1146,6 +1185,111 @@ impl From<MediaType> for MediaKind {
     }
 }
 
+#[allow(clippy::unusual_byte_groupings)]
+fn finalize_packet(
+    next: NextPacket,
+    now: Instant,
+    exts: &Extensions,
+    twcc: &mut u64,
+    mid: Mid,
+    buf: &mut Vec<u8>,
+) -> (RtpHeader, SeqNo) {
+    let (mut header, seq_no, orig_seq_no) = match next {
+        NextPacket::ResendOrRegular {
+            pt,
+            pkt,
+            ssrc,
+            seq_no,
+            orig_seq_no,
+        } => {
+            let header = RtpHeader {
+                payload_type: pt,
+                sequence_number: *seq_no as u16,
+                timestamp: pkt.meta.rtp_time.numer() as u32,
+                ssrc,
+                ext_vals: pkt.meta.ext_vals,
+                marker: pkt.last,
+                ..Default::default()
+            };
+
+            (header, seq_no, orig_seq_no)
+        }
+        NextPacket::Padding {
+            pt, ssrc, seq_no, ..
+        } => {
+            let header = RtpHeader {
+                payload_type: pt,
+                sequence_number: *seq_no as u16,
+                timestamp: 0,
+                ssrc,
+                ..Default::default()
+            };
+
+            (header, seq_no, None)
+        }
+    };
+
+    // We can fill out as many values we want here, only the negotiated ones will
+    // be used when writing the RTP packet.
+    //
+    // These need to match `Extension::is_supported()` so we are sending what we are
+    // declaring we support.
+    header.ext_vals.abs_send_time = Some(MediaTime::new_ntp_time(now));
+    header.ext_vals.mid = Some(mid);
+    header.ext_vals.transport_cc = Some(*twcc as u16);
+    *twcc += 1;
+
+    buf.resize(2000, 0);
+    let header_len = header.write_to(buf, exts);
+    assert!(header_len % 4 == 0, "RTP header must be multiple of 4");
+    header.header_len = header_len;
+
+    let mut body_out = &mut buf[header_len..];
+
+    // For resends, the original seq_no is inserted before the payload.
+    let mut original_seq_len = 0;
+    if let Some(orig_seq_no) = orig_seq_no {
+        original_seq_len = RtpHeader::write_original_sequence_number(body_out, orig_seq_no);
+        body_out = &mut body_out[original_seq_len..];
+    }
+
+    match next {
+        NextPacket::ResendOrRegular { pkt, .. } => {
+            let body_len = pkt.data.len();
+            body_out[..body_len].copy_from_slice(&pkt.data);
+
+            // pad for SRTP
+            let pad_len = RtpHeader::pad_packet(
+                &mut buf[..],
+                header_len,
+                body_len + original_seq_len,
+                SRTP_BLOCK_SIZE,
+            );
+
+            buf.truncate(header_len + body_len + original_seq_len + pad_len);
+        }
+        NextPacket::Padding { len, .. } => {
+            let rounded_len = if len as usize % SRTP_BLOCK_SIZE == 0 {
```

Comment:
Maybe generating this padding is a bit to low-level for this crate. Perhaps we should have

```rust
RtpHeader::generate_padding_packet(&mut buf[..], len, SRTP_BLOCK_SIZE)
```

or something like it